testutils/resourcebuilder/machine/v1beta1/machineset.go

```diff
@@ -88,32 +142,123 @@ func (m MachineSetBuilder) AsWorker() MachineSetBuilder {
 		WithLabel(machineSetMachineTypeLabelName, "worker")
 }
 
+// WithAnnotations sets the annotations for the machineSet on the machineSet builder.
+// DEPRECATED: use WithMachineSetAnnotations() instead.
```

Comment:
Why is this special and prefixing the labels/annotations with MachineSet? I wouldn't expect that on other resources.

Do we do something similar on CPMS?

Reply:
We do have MachineSet (top level annotations/labels on the resource), and we then have the machineTemplate ones, which are then propagated to the children Machines. I wanted to have a way to differentiate the two, as in real MachineSet they could be different right?

Reply:
I agree that these should be different, but wondered why we don't just call these top level ones `WithLabels` and `WithAnnotations`, since that's what we do on other objects isn't it?

The ones within the spec I'm happy to be named differently, but thought we should be consistent on the top level object meta ones

Reply:
Yeah I did that to avoid confusion, but I agree having consistency is also desireable.  In that case we will need to set both labels to the same for AsWorker() now.

Addressed that in the latest commit.

Also testcases that currently use MachineSet.WithLabels() will need updating as now  WithLabels() will only update ms.Labels and not also ms.Spec.Selector.MatchLabels. (e.g. https://github.com/search?q=repo%3Aopenshift%2Fmachine-api-operator%20WithLabel&type=code )